### PR TITLE
Ensure stop on entry reason is `entry`. Fixes #1387

### LINKED ISSUE #1387: Stop on entry should return stopped reason as 'entry'
This is needed to support F5/F11 scenarios in VS.

### LINKED ISSUE #1418: DAP: Stop on entry should be ported to pydevd
Blocked by:

https://github.com/microsoft/ptvsd/issues/1387
https://github.com/microsoft/ptvsd/issues/1413

### PR DESCRIPTION
**Pull request by karthiknadig**
Fixes #1387
Fixes #1418

### CODE REVIEW

**Comment by fabioz** [comment]
To update `PyDBAdditionalThreadInfo` you should change only `pydevd_cython.pxd` and the `__slots__` / constructor initialization in `pydevd_additional_thread_info_regular.py`, then, run the command below to update the related cython files (updating the path to `X:\ptvsd_workspace` to match your installation).

```
cd /D X:\ptvsd_workspace\ptvsd\src\ptvsd\_vendored\pydevd
set PYTHONPATH=X:\ptvsd_workspace\ptvsd\src\ptvsd\_vendored\pydevd
C:\bin\Miniconda\envs\py36_64\python build_tools\build.py
```

Note that you should also commit the `.c` files.

**Comment by karthiknadig** [comment]
With VS 2019 the existing build.py fails with can't find `vcvarsall` I temporarily changed that to use the 2019 version to build the .c files.

**Comment by fabioz** [comment]
The build is failing because it's missing `pydevd_stop_on_entry` in `pydevd_cython.pxd`.

**Comment by karthiknadig** [comment]
Not sure why it is not picking up the newly generated .c files. Is there any issue in the generated files?

**Comment by karthiknadig** [comment]
Ah... I see the commit was missing a file. Updated it.

**Comment by karthiknadig** [comment]
This is ready for review.  #1413 is due to `debug` entry point used by VS. This can be addressed separately. the workaround is to turn of stop on entry and use breakpoint instead (in VS). This issue exists in the shipped product. so we can check this change in.

File: src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_api.py [hunk 1]
```diff
@@ -554,3 +555,8 @@ def custom_dont_trace_external_files(abs_path):
         if reset_caches:
             py_db.clear_dont_trace_start_end_patterns_caches()
 
+    def stop_on_entry(self):
+        main_thread = pydevd_utils.get_main_thread()
```
Comment (fabioz) [inline src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_api.py]: I think it'd be nice to check if `main_thread` is not `None` here (and if it is show a `pydev_log.critical()` to the user) just to be on the safe side.

**Pull request merged**